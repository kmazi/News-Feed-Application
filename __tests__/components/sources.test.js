import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from 'enzyme';
import ReactRenderer from 'react-test-renderer';
import Sources from './../../app/components/sources.jsx';

describe('The sources component', () => {
  it('must render without throwing error', () => {
    const div = document.createElement('div');
    ReactDom.render(<Sources />, div);
  });

  it('should show loading... initially before loading from api', () => {
    const anchor = shallow(<Sources />);
    expect(anchor.text().includes('loading...')).toBe(true);
  });

  it('should render all component correctly', () => {
    const component = ReactRenderer.create(<Sources/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
