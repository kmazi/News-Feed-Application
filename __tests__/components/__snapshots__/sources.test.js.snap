// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The sources component should render all component correctly 1`] = `
<div
  className="col-md-3"
  id="news-source"
>
  <h3
    className=""
  >
    News Channels
  </h3>
  <div>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="search news sources"
    />
    <h4>
      loading...
    </h4>
  </div>
</div>
`;
